shared/util.go

```diff
@@ -1237,3 +1239,21 @@ func JoinUrls(baseUrl, p string) (string, error) {
 	u.Path = path.Join(u.Path, p)
 	return u.String(), nil
 }
+
+// GetStableRandomValueFromList returns a stable random value from a given list.
```

Comment:
@tomponline can we use that for the other cases you already implemented?

Reply:
@stgraber it looks like it could be used in `autoCreateCustomVolumeSnapshotsTask` and `addChassisGroupEntry`, as they pick from lists. However the other two use-cases are for generating MAC addresses. Searching for `fnv.New64a` shows all the current use-cases.

Also the seed in the other use-cases is a string not a single int.

The common parts are:

```go
hash := fnv.New64a()
_, err := io.WriteString(hash, seedStr)
if err != nil {
    return err
}

r := mrand.New(mrand.NewSource(int64(hash.Sum64())))
```

If this was refactored into two functions, firstly `GetStableRandomGenerator(seed string) *rand.Rand ` that just took a string seed and returned a preconfigured random generator, then it could be used for all use-cases, and then optionally a 2nd function `GetRandomNumber(r *rand.Rand, max int64) int64` this could then be used as so:

```go
r, err := GetStableRandomGenerator("someSeed")
if err != nil {
     return err
}

fmt.Printf("%d", myList[GetRandomNumber(r, len(myList))])
```

Reply:
@monstermunchkin please could you move this part (below) into its own function called `GetStableRandomGenerator(seed string) *rand.Rand`

```
hash := fnv.New64a()
_, err := io.WriteString(hash, seedStr)
if err != nil {
    return err
}

return mrand.New(mrand.NewSource(int64(hash.Sum64())))
```

Then you can use it in `GetStableRandomValueFromList` and I can use it later in the other stable random number generator scenarios. Thanks

Reply:
Done.